docs/resources/guidelines/guideline_containers.md

```diff
@@ -1 +1,46 @@
-# Guideline - Containers
\ No newline at end of file
+# Guideline - Containers
+
+### 1. Securing Images:
+- `Include your application in a container image`: A container image holds a subset of the OS along with the application. Each library/tool you pull into the image poses a potential threat. To mitigate these threats, you need to include the application within the container image.
+- `Include as little as possible`: Remove all components that aren't required by the application. This can help you reduce the attack surface.
+- `Use trustworthy images`: If you are not creating the image from scratch, you should choose images that are trusted official images hosted on sites like **Docker Hub**.
+
+### 2. Securing Registries
+- `Use access control`: If you have your own private registry, you need to establish access controls that define exactly who can access and publish images and who cannot perform these actions; which prevents unauthorized parties from changing these images.
+- `Sign your images`: Signatures help track images to the people who signed them. This makes it difficult to substitute the signed image for a compromised one. The **Docker Content Trust** mechanism provides information about signing images. You can also use **Notary**, an open-source tool that helps you sign and verify images.
+- `Scan your images`: Use vulnerability scanners like **Snyk** to check if the image is having any known vulnerability.
+
+### 3. Securing Deployment
+- `Secure the target environment`: You can do this by hardening the underlying host operating system. You can also establish firewall and VPC rules or create special accounts that limit access.
+- `Use an orchestration platform`: These systems usually provide secure API endpoints as well as role-based access control, which can help minimize the risk of unauthorized access.
+- `Use immutable deployments`: This involves creating an instance image during your build steps. Your deployment can then use this image to create new instances.
+
+### 4. Securing Container Runtime
+- Create separate virtual networks for your containers.
+- `Apply the principle of least privilege`: Allow connectivity only between containers that truly need it.
+- `Expose only the ports that serve the application`: Do not expose any other ports, except for SSH. Apply this principle to containers as well as the underlying machines.
```

Comment:
You do generally NOT want to expose port 22 (SSH) on any internet facing containers!

Reply:
Where should I add the link?
At the beginning or end or where I have mentioned Snyk?